src/app/loadout-builder/preProcessFilter.ts

```diff
@@ -99,3 +102,43 @@ export function matchLockedItem(item: DimItem, lockedItem: LockedItemType) {
       return item.id === lockedItem.item.id;
   }
 }
+
+export function getTotalBaseStatsWithMasterwork(item: DimItem, assumeMasterwork: boolean | null) {
+  const stats = _.keyBy(item.stats, (stat) => stat.statHash);
+  const baseStats = {};
+
+  for (const statHash of statValues) {
+    baseStats[statHash] = stats[statHash]?.base || 0;
+  }
+
+  // Checking energy tells us if it is Armour 2.0
+  if (item.isDestiny2() && item.sockets && item.energy) {
+    // If not assume masterwork add on mw stats as they can't be removed.
+    if (!assumeMasterwork) {
+      const masterworkSocketHashes = getMasterworkSocketHashes(
+        item.sockets,
+        DestinySocketCategoryStyle.EnergyMeter
+      );
+
+      for (const socket of item.sockets.sockets) {
+        const plugHash = socket?.plug?.plugItem?.hash ?? NaN;
+
+        if (socket.plug?.stats && !masterworkSocketHashes.includes(plugHash)) {
+          for (const statHash of statValues) {
+            if (socket.plug.stats[statHash]) {
+              baseStats[statHash] += socket.plug.stats[statHash];
+            }
+          }
+        }
+      }
+    }
+
+    if (assumeMasterwork) {
```

Comment:
This can be `else`, and it should be inverted to consider `assumeMasterwork` first